### PR TITLE
Fix update by rebase being wrongly disabled by protected base branch

### PR DESCRIPTION
**Pull request by brechtvl**
The branch this is force pushing to is the head branch in the head repo, so it should be checking if that is protected, not the base.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: services/pull/update.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func IsUserAllowedToUpdate(ctx context.Context, pull *issues_model.PullRequest,
 		BaseBranch: pull.HeadBranch,
 	}
 
-	pb, err := git_model.GetFirstMatchProtectedBranchRule(ctx, pull.BaseRepoID, pull.BaseBranch)
+	pb, err := git_model.GetFirstMatchProtectedBranchRule(ctx, pull.HeadRepoID, pull.HeadBranch)
```
Comment (lunny) [inline services/pull/update.go]: It's better to use `pr.BaseRepoID` and `pr.BaseBranch`.

**Review by wolfogre — APPROVED** [review]

File: services/pull/update.go (hunk 1 above)
Comment (wolfogre) [inline services/pull/update.go]:
```suggestion
	pb, err := git_model.GetFirstMatchProtectedBranchRule(ctx, pr.BaseRepoID, pr.BaseBranch)
```

**Comment by wolfogre** [comment]
@GiteaBot

**Review by delvh — APPROVED** [review]

**Comment by yardenshoham** [comment]
Please send backport

**Comment by zeripath** [comment]
I don't believe that this is a bug in the 1.18 code therefore this is not backportable.

**Pull request merged**